### PR TITLE
Claim improvements

### PR DESCRIPTION
**Pull request by sheryew**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+        String finalString = "";
+
+        for (String key: references.keySet()) {
+            List<Index> indexList = references.get(key);
+            String tempStr = "You are viewing " + key + ":\n";
+            for (int i = 0; i < indexList.size(); i++) {
+                int listIndex = indexList.get(i).getZeroBased();
+                if (listIndex >= lastShownList.size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
```
Comment (remuslum) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:62]: Can consider adding more test cases for failure scenarios in view command

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                }
+                Person personToEdit = lastShownList.get(indexList.get(i).getZeroBased());
+                HashMap<String, String> personRelevantInfo = generateArrInfo(personToEdit);
+                int index = indexList.get(i).getOneBased();
+                if (Objects.equals(key, NAME_IDENTIFIER)) {
+                    tempStr += String.format("%s. %s.\n", index, personRelevantInfo.get(NAME_IDENTIFIER));
```
Comment (remuslum) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:68]: I think here can use assert or j include test cases method in the ViewCOmmandTest.java file

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java [hunk 3]
```diff
@@ -145,4 +151,41 @@ Prefix getPrefix() {
...
+     * @param argsString User's input.
+     * @return ArgumentMultimap object.
+     * @throws ParseException if duplicate prefixes are found.
+     */
+    public static ArgumentMultimap viewTokenize(String argsString) throws ParseException {
+        List<String> extractedPrefixes = extractPrefixes(argsString);
+
+        Set<String> uniquePrefixes = new HashSet<>(extractedPrefixes);
+        if (uniquePrefixes.size() != extractedPrefixes.size()) {
+            throw new ParseException("Duplicate prefix detected in input.");
```
Comment (remuslum) [inline src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java:185]: Same issue here, can consider adding more test cases for failure in ArgumentTokenizerTest.java

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -507,6 +507,27 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1a2. HR Insight shows all the attributes (Prefixes) it can display for employee(s).
+
+  Use case ends.
+
+* 1b. User didn't provide any attribute that it wants to view.
+    * 1b1. HR Insight shows an error message.
+    * 1b2. HR Insights shows all the attributes (Prefixes) it can display for employee(s).
+
+  Use case ends.
+
```
Comment (marcellaantania) [inline docs/DeveloperGuide.md:530]: What happens when the user writes "view b/1,2 s/3" for example? do the indices have to match?

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+     *
+     * @param personInfo HashMap containing Person's primary attributes.
+     * @param index Integer which represents index from list.
+     * @param identifier String which represents what attribute to view from.
+     * @return String representation of relevant information.
+     */
+    public String generateNiceStr(HashMap<String, String> personInfo, int index, String identifier) {
+        String result = personInfo.get(identifier);
+        String name = personInfo.get(NAME_IDENTIFIER);
+        return String.format("%s. %s's %s is %s.\n", index, name, identifier, result);
```
Comment (marcellaantania) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java:120]: when the command is "view b/1,2 s/1" will the list numbering be 1. 2. 1.? Not sure what index and indexlist refer to in this code

**Review by marcellaantania — COMMENTED** [review]
I know that this has taken you a lot of effort to make (the chaining and all) but wouldn't it be less confusing to allow just one view at a time? Meaning users can only view one attribute of one person at a time. Since we also have find, it is relatively easier to use find instead of view if users wish to see multiple attributes. 

Alternatively, we can just do "view INDEX" to just show all attributes of the specified employees altogether. Maybe allow index chaining here.  I feel like it'll be easier for us and for the users as well. So users can do find to find names and do view if they know the indices.

**Comment by nixonwidjaja** [comment]
@sheryew I don't see any specific use cases here. I agree with @marcellaantania that view INDEX and find NAME is better than cherrypicking attributes from different indices.

File: src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java [hunk 6]
```diff
@@ -1,8 +1,10 @@
 package seedu.address.logic.parser;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+import java.util.*;
```
Comment (remuslum) [inline src/main/java/seedu/address/logic/parser/ArgumentTokenizer.java]: Try not to use * imports, think it affects our code quality marks

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 7]
```diff
@@ -8,17 +8,7 @@
...
-import seedu.address.logic.commands.ClearCommand;
-import seedu.address.logic.commands.Command;
-import seedu.address.logic.commands.DeleteCommand;
-import seedu.address.logic.commands.EditCommand;
-import seedu.address.logic.commands.ExitCommand;
-import seedu.address.logic.commands.FindCommand;
-import seedu.address.logic.commands.HelpCommand;
-import seedu.address.logic.commands.LeaveCommand;
-import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.commands.*;
```
Comment (remuslum) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java]: same issue with the * import

Comment (sheryew) [reply src/main/java/seedu/address/logic/parser/AddressBookParser.java]: weirdly, it is gone HAHA

**Review by nixonwidjaja — APPROVED** [review]

**Pull request merged**